internal/services/cosmos/cosmosdb_account_resource.go

```diff
@@ -1147,31 +1147,13 @@ func resourceCosmosDbAccountDelete(d *pluginsdk.ResourceData, meta interface{})
 		return err
 	}
 
-	if _, err := client.Delete(ctx, id.ResourceGroup, id.Name); err != nil {
+	future, err := client.Delete(ctx, id.ResourceGroup, id.Name)
+	if err != nil {
 		return fmt.Errorf("deleting CosmosDB Account %q (Resource Group %q): %+v", id.Name, id.ResourceGroup, err)
 	}
 
-	// the SDK now will return a `WasNotFound` response even when still deleting
```

Comment:
have you confirmed with the service team this is no longer required? LRO's don't always fix this